ui.py

```diff
@@ -0,0 +1,200 @@
+        output_folder = gr.Textbox(label="Output Folder Path", value=cmd_args.output_folder, interactive=False)
+
+    with gr.Row():
+        newline_mode = gr.Radio(choices=["single", "double"], label="Newline Mode", value="double")
+        remove_endnotes = gr.Checkbox(label="Remove Endnotes", value=False)
+        output_text = gr.Checkbox(label="Output Text", value=False)
+        log = gr.Dropdown(choices=["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"], label="Log Level", value="INFO")
+
+
+    # Azure TTS Tab
+    with gr.Tab("Azure TTS"):
+        azure_tts_key = gr.Textbox(label="Azure TTS Key", value="")
+        azure_tts_region = gr.Textbox(label="Azure TTS Region", value="")
+        voice_name = gr.Textbox(label="Azure Voice Name", value="en-US-GuyNeural")
+        break_duration = gr.Textbox(label="Break Duration (ms)", value="1250")
+        output_format = gr.Dropdown(choices=["audio-16khz-32kbitrate-mono-mp3", "audio-16khz-64kbitrate-mono-mp3", "audio-16khz-128kbitrate-mono-mp3", "audio-24khz-48kbitrate-mono-mp3", "audio-24khz-96kbitrate-mono-mp3", "audio-24khz-160kbitrate-mono-mp3", "audio-48khz-96kbitrate-mono-mp3", "audio-48khz-192kbitrate-mono-mp3"], label="Output Format", value="audio-24khz-48kbitrate-mono-mp3")
+
+    # OpenAI TTS Tab
+    with gr.Tab("OpenAI TTS"):
+        openai_api_key = gr.Textbox(label="OpenAI API Key", value="")
+        openai_model = gr.Dropdown(choices=["tts-1", "tts-1-hd"], label="OpenAI Model", value="tts-1")
+        openai_voice = gr.Dropdown(choices=["alloy", "echo", "fable", "onyx", "nova", "shimmer"], label="OpenAI Voice", value="alloy")
+        openai_format = gr.Dropdown(choices=["mp3", "opus", "aac", "flac"], label="OpenAI Format", value="mp3")
+
+    # Submit & Stop Button
+    with gr.Row():
+        submit_button = gr.Button("Convert to Audiobook", variant="primary")
+        stop_button = gr.Button("Stop", variant="stop")
+    log_textarea = gr.TextArea(label="Log", interactive=False, lines=10)
+    file_list = gr.File(label="Download Audiobook", file_count="multiple", interactive=False)
+
+    input_file.upload(conversion.preview_book, inputs=[input_file], outputs=[chapter_end])
+    
+    submit_button.click(
+        conversion.convert_epub_to_audiobook, 
+        inputs=[
+            input_file,
+            tts, log, language, newline_mode, chapter_start, chapter_end,
+            output_text, remove_endnotes,
+            azure_tts_key, azure_tts_region,
+            voice_name, break_duration, output_format,
+            openai_api_key,
+            openai_model, openai_voice, openai_format],
+        outputs=[],
+    )
+
+    stop_button.click(conversion.stop_subprocess)
+    ui.load(utils.read_log, inputs=None, outputs=log_textarea, every=1)
+    ui.load(conversion.list_files, inputs=None, outputs=[file_list], every=1)
+
+ui.queue().launch(server_name=cmd_args.listen, server_port=cmd_args.port, share=cmd_args.share)
```

Comment:
No new line at the end ahahahha

Overall looks awesome, though it was implemented before major project refactoring.
Could you please update this PR to match current project version and I would love to merge it in